meilisearch-http/src/routes/tasks.rs

```diff
@@ -180,26 +180,79 @@ impl From<Details> for DetailsView {
     }
 }
 
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct TaskDateQuery {
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "time::serde::rfc3339::option::serialize",
+        deserialize_with = "rfc3339_date_or_datetime::deserialize"
+    )]
+    after_enqueued_at: Option<OffsetDateTime>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "time::serde::rfc3339::option::serialize",
+        deserialize_with = "rfc3339_date_or_datetime::deserialize"
+    )]
+    before_enqueued_at: Option<OffsetDateTime>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "time::serde::rfc3339::option::serialize",
+        deserialize_with = "rfc3339_date_or_datetime::deserialize"
+    )]
+    after_started_at: Option<OffsetDateTime>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "time::serde::rfc3339::option::serialize",
+        deserialize_with = "rfc3339_date_or_datetime::deserialize"
+    )]
+    before_started_at: Option<OffsetDateTime>,
+
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "time::serde::rfc3339::option::serialize",
+        deserialize_with = "rfc3339_date_or_datetime::deserialize"
+    )]
+    after_finished_at: Option<OffsetDateTime>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        serialize_with = "time::serde::rfc3339::option::serialize",
+        deserialize_with = "rfc3339_date_or_datetime::deserialize"
+    )]
+    before_finished_at: Option<OffsetDateTime>,
+}
+
 #[derive(Deserialize, Debug)]
 #[serde(rename_all = "camelCase", deny_unknown_fields)]
 pub struct TasksFilterQuery {
     #[serde(rename = "type")]
-    type_: Option<CS<StarOr<Kind>>>,
+    kind: Option<CS<StarOr<Kind>>>,
+    uid: Option<CS<u32>>,
     status: Option<CS<StarOr<Status>>>,
-    index_uid: Option<CS<StarOr<IndexUid>>>,
+    index_uid: Option<CS<StarOr<String>>>,
```

Comment:
Why have we changed that?

Reply:
See https://github.com/meilisearch/meilisearch/pull/2932#discussion_r999520318 , and also for consistency with the way I had implemented it for TaskDeletion